ga4gh/avrotools.py

```diff
@@ -129,309 +92,419 @@ def _ensureProtocolSubclass(self):
-
-    @classmethod
-    def _doubleGenerator(cls, min_=None, max_=None):
-        """
-        Creates a generator which returns a floating point number between min
-        and max.
-        """
-        if min_ is None:
-            min_ = cls.LONG_MIN_VALUE
-        if max_ is None:
-            max_ = cls.LONG_MAX_VALUE
-        while True:
-            yield random.uniform(min_, max_)
-
-    @classmethod
-    def _stringGenerator(cls, length=10, characters=string.printable):
-        """
-        Creates a generator which returns a string with the supplied
-        character set (defaults to printable characters).
-        """
-        gen = cls._valueFromSetGenerator(characters)
-        while True:
-            yield "".join(itertools.islice(gen, length))
+    def handleLong(self, min_=LONG_MIN_VALUE, max_=LONG_MAX_VALUE):
+        return random.randint(min_, max_)
 
-    @classmethod
-    def _unionGenerator(cls, generators):
-        """
-        Creates a generator that returns a value from a randomly chosen
-        generator from the supplied list.
-        """
-        while True:
-            yield random.choice(generators).next()
+    def handleFloat(self, min_=INT_MIN_VALUE, max_=INT_MAX_VALUE):
+        return random.uniform(min_, max_)
 
-    @classmethod
-    def _arrayGenerator(cls, generator, length=10):
-        """
-        Creates a generator that returns an an array of values taken from the
-        supplied generator.
-        """
-        while True:
-            yield list(itertools.islice(generator, length))
+    def handleDouble(self, min_=LONG_MIN_VALUE, max_=LONG_MAX_VALUE):
+        return random.uniform(min_, max_)
 
-    @classmethod
-    def _mapGenerator(cls, valueGen, size=10):
-        """
-        Creates a generator that returns an a dictionary with keys generated
-        by keyGen and values generated by valueGen.
-        """
-        keyGen = cls._stringGenerator()
-        while True:
-            keys = list(itertools.islice(keyGen, size))
-            values = list(itertools.islice(valueGen, size))
-            yield dict(zip(keys, values))
-
-    @classmethod
-    def _recordGenerator(self, recordSchema):
-        """
-        Creates a generator which returns a dictionary where keys and values
-        are determined by the recordSchema. The recordSchema is a map of the
-        records keys to a generator of valid values for that key.
-        """
-        while True:
-            record = {}
-            for key, gen in recordSchema.iteritems():
-                record[key] = gen.next()
-            yield record
+    def handleFixed(self, schema):
+        return self.handleString(schema.size)
 
-    def __init__(self, class_):
-        self.class_ = class_
-        self.schema = class_.schema
+    def handleEnum(self, schema):
+        return random.choice(schema.symbols)
+
+    def handleArray(self, schema, length=10):
+        toReturn = []
+        for _ in xrange(length):
```

Comment:
`xrange` goes away in Python 3, so we should just stick to `range`. The performance issues aren't worth worrying about in this context.